core/src/main/java/hudson/model/DownloadService.java

```diff
@@ -397,7 +397,11 @@ private FormValidation load(String json, long dataTimestamp) throws IOException
         public FormValidation updateNow() throws IOException {
             List<JSONObject> jsonList = new ArrayList<>();
             boolean toolInstallerMetadataExists = false;
-            for (String site : getUrls()) {
+            for (UpdateSite updatesite : Jenkins.getActiveInstance().getUpdateCenter().getSiteList()) {
+                String site = updatesite.getMetadataUrlForDownloadable(url);
+                if (site == null) {
+                    return FormValidation.warning("One of the update site does not look like an update center");
```

Comment:
>Grammar: updates sites

@daniel-beck that smells wrong to me...   `update sites` (the singular being `update site` not `updates site` even if the singular as used thoughout Jenkins would potentially be wrong)